packages/image_picker/image_picker/example/lib/main.dart

```diff
@@ -37,8 +37,10 @@ class MyHomePage extends StatefulWidget {
 
 class _MyHomePageState extends State<MyHomePage> {
   PickedFile? _imageFile;
+  List<PickedFile>? _imageFileList;
```

Comment:
For the example it is not really relevant whether you show 1 image or multiple images. 

You can change the `_imageFile` into a setter and set the value in the list.
```suggestion
  set _imageFile (PickedFile? value) {
    _imageFileList = value == null ? null : [value];
  }
  List<PickedFile>? _imageFileList;
```

or change the line in `_displayPickImageDialog`:
```dart
_imageFileList = pickedFile == null ? null :  [pickedFile];
```

You can then remove the `isMultiImage` and the widget for displaying 1 image. The widget for 1 image is exactly the same as that for displaying multiple, so it doesn't really add anything.